python-lib/dku_data_processing/format.py

```diff
@@ -31,6 +32,62 @@ def wkt_parser(wkt_point):
         return result[0]
 
 
+def extract_df(df, detail, filters, geopoint, tooltips):
+    """
+
+    Extract and select only necessary data to send to front-end.
+    Handle the filtering, normalizing, parsing and the tooltip.
+
+    :param detail:
+    :param df:
+    :param filters:
+    :param geopoint:
+    :param tooltips:
+    """
+    # Handle filtering
```

Comment:
I feel that you could turn each commented block into a function doing what your comment says

Reply:
Will leave this out of this PR but duly noted